src/NServiceBus.Core/Transports/Sending.cs

```diff
@@ -1,53 +1,60 @@
+#pragma warning disable 1591
 namespace NServiceBus
 {
-    using System;
-    using System.Threading;
-    using System.Threading.Tasks;
-    using Features;
+    using ObjectBuilder;
+    using Routing;
+    using Settings;
     using Transport;
 
-    class Sending : Feature
+    public class TransportComponent
     {
-        public Sending()
+        public TransportComponent(TransportInfrastructure transportInfrastructure)
         {
-            EnableByDefault();
+            TransportInfrastructure = transportInfrastructure;
         }
 
-        protected internal override void Setup(FeatureConfigurationContext context)
-        {
-            var transport = context.Settings.Get<OutboundTransport>();
-            var lazySendingConfigResult = new Lazy<TransportSendInfrastructure>(() => transport.Configure(context.Settings), LazyThreadSafetyMode.ExecutionAndPublication);
-            context.Container.ConfigureComponent(c =>
-            {
-                var dispatcher = lazySendingConfigResult.Value.DispatcherFactory();
-                return dispatcher;
-            }, DependencyLifecycle.SingleInstance);
+        public IDispatchMessages Dispatcher { get; set; }
```

Comment:
I don't think it's wrong to use the raw dispatcher at that point but one valid argument by szymon is, that feature setups are not async and therefore you'd have to fire & forget or to use blocking calls if you want to send. This is of course an indication that you shouldn't use it in the setup but that's kind of a hidden contract.